src/mds/MDSAuthCaps.cc

```diff
@@ -373,6 +373,80 @@ bool MDSAuthCaps::parse(CephContext *c, std::string_view str, ostream *err)
   }
 }
 
+bool MDSAuthCaps::merge(MDSAuthCaps newcap)
+{
+  assert(newcap.grants.size() == 1);
+  auto ng = newcap.grants[0];
+
+  for (auto& g : grants) {
+    if (g.match.fs_name == ng.match.fs_name && g.match.path == ng.match.path) {
+      if (g.match.uid == ng.match.uid && g.match.gids == ng.match.gids &&
```

Comment:
Next, tests caps for UID and GID myself, choose right behaviour of fs authorize in this case. Added tests and docs accordingly.

Reply:
Also added release note.